odoo/addons/base/tests/test_ir_cron.py

```diff
@@ -261,6 +242,72 @@ def test_cron_process_job(self):
                     ('call_at', '<=', fields.Datetime.now())]
                 ), 0)
 
+    def test_cron_recall(self):
+        def run(self):
+            if not self.env['ir.cron.progress'].search_count([('progress', '=', 1)]):
+                self.env['ir.cron']._log_progress(1, 1)
+
+        with self.capture_triggers() as capture:
+            self.cron._trigger()
+        self.cron.flush_recordset()
+        capture.records.flush_recordset()
```

Comment:
Why don't you simply `self.env.flush_all()` ?
If `cron._trigger()` is overridden and updates more records, your test will do weird stuff.

Reply:
It's what was mostly used in the other's test (with triggers), suggestion makes sense will also apply to other tests (who use trigger)